File: store/src/test/java/com/nytimes/android/external/store/NoNetworkTest.java
package com.nytimes.android.external.store;

import com.nytimes.android.external.store.base.impl.BarCode;
import com.nytimes.android.external.store.base.impl.Store;
import com.nytimes.android.external.store.base.impl.StoreBuilder;

import org.junit.Before;
import org.junit.Test;

import rx.Observable;

/**
 * Created by 206847 on 5/3/17.
 */

public class NoNetworkTest {

    private static final RuntimeException EXCEPTION = new RuntimeException();
    private Store<Object, BarCode> store;

    @Before
    public void setUp() {
        store = StoreBuilder.barcode()
                .fetcher(barcode -> Observable.error(EXCEPTION))
                .open();
    }

    @Test
    public void testNoNetwork() throws Exception {
        store.get(new BarCode("test", "test"))
                .test()
                .awaitTerminalEvent()
                .assertError(EXCEPTION);
    }
}

File: app/src/test/java/com/nytimes/android/sample/NoNetworkTest.kt
package com.nytimes.android.sample

import com.nytimes.android.external.store3.base.impl.BarCode
import com.nytimes.android.external.store3.base.impl.StoreBuilder
import kotlinx.coroutines.runBlocking
import org.junit.Test

class NoNetworkTest {
    private val store = StoreBuilder.barcode<Any>()
            .fetcher { throw EXCEPTION }
            .open()

    @Test(expected = java.lang.Exception::class)
    fun testNoNetwork() = runBlocking<Unit> {
        store.get(BarCode("test", "test"))
    }

    companion object {
        private val EXCEPTION = RuntimeException()
    }
}
